evm/src/generation/prover_input.rs

```diff
@@ -292,6 +304,55 @@ impl<F: Field> GenerationState<F> {
             ))
         }
     }
+
+    /// Returns the a pointer to an element in the list whose value val is such that
+    /// val <= addr < next_val and addr is the top of the stack.
+    fn run_next_addresses_insert(&mut self) -> Result<U256, ProgramError> {
+        let addr = stack_peek(self, 0)?;
+        for (curr_ptr, next_addr, _) in self.get_addresses_access_list()? {
+            if next_addr > addr {
+                return Ok((Segment::AccessedAddresses as usize + curr_ptr).into());
+            }
+        }
+        Ok((Segment::AccessedAddresses as usize).into())
+    }
+
+    /// Returns the a pointer to an element in the list whose value val is such that
+    /// val < addr == next_val and addr is the top of the stack. If the element is not
+    /// in the list returns loops forever
+    fn run_next_addresses_remove(&mut self) -> Result<U256, ProgramError> {
+        let addr = stack_peek(self, 0)?;
+        for (curr_ptr, next_addr, _) in self.get_addresses_access_list()? {
+            if next_addr == addr {
+                return Ok((Segment::AccessedAddresses as usize + curr_ptr).into());
```

Comment:
I would like to use the abstraction of pointers, because these offsets/virtual addresses in `Segment::AccessedAddresses` are interpreted as pointers to nodes in the linked list of accessed address/storages keys. I'm also using ptr on the kernel side.